state/protocol/util/testing.go

```diff
@@ -72,6 +74,30 @@ func RunWithBootstrapState(t testing.TB, rootSnapshot protocol.Snapshot, f func(
 	})
 }
 
+// RunWithFollowerProtocolStateAndResults will insert some results into results db. Used to test sealing segment edge cases.
+func RunWithFollowerProtocolStateAndResults(t testing.TB, rootSnapshot protocol.Snapshot, exeResults []*flow.ExecutionResult, f func(*badger.DB, *pbadger.FollowerState)) {
```

Comment:
I'm not sure it's necessary to duplicate this utility just to also add results. Could we do something like:
```go
// use the generic test utility
RunWithFollowerProtocolState(..., func(db *badger.DB, state *bprotocol.FollowerState) {

    // add the results needed for the specific test case
    for _, result := range exeResults {
        db.Update(operation.InsertExecutionResult(result))
    }

    // ...
}
```

Reply:
nice thanks